### PR TITLE
Move siacoin outputs map to disk

### PR DESCRIPTION
**Pull request by triazo**
Mostly straightforward changes, had no hiccups when implementing. This is on top of moving file contracts to disk so look at that first.

### CODE REVIEW

**Comment by DavidVorick** [comment]
looks like this needs to be rebased on top of the other one.

**Comment by DavidVorick** [comment]
test timeout

**Comment by triazo** [comment]
Seems to be better now. It seems that Travis was just being really slow, as the timeout occurred on creating the consensus set tester object, which every other test case does.

**Comment by DavidVorick** [comment]
I am cautious to think that Travis is at fault. It's weird that it would take 3 minutes to make a consensus set tester object, and difficult to blame that on Travis.

**Comment by DavidVorick** [comment]
there are now merge conflicts due to my recent PR being merged.

File: modules/consensus/accept_test.go [hunk 1]
```diff
@@ -439,16 +450,19 @@ func (cst *consensusSetTester) testSpendSiacoinsBlock() error {
...
-	for id, output := range cst.cs.siacoinOutputs {
+	cst.cs.db.forEachSiacoinOutputs(func(id types.SiacoinOutputID, output types.SiacoinOutput) {
 		if id == outputID {
 			if found {
-				return errors.New("output found twice")
+				err = errors.New("output found twice")
 			}
 			if output.Value.Cmp(txnValue) != 0 {
-				return errors.New("output has wrong value")
+				err = errors.New("output has wrong value")
```
Comment (DavidVorick) [inline modules/consensus/accept_test.go]: You want to let it finish iterating even though you are going to return an error at the end anyway?

File: modules/consensus/accept_test.go [hunk 2]
```diff
@@ -1495,14 +1513,17 @@ func TestComplexForking(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
+	defer cst1.cs.Close()
```
Comment (DavidVorick) [inline modules/consensus/accept_test.go]: did you mean to call closeCst()?

File: modules/consensus/accept_test.go [hunk 3]
```diff
@@ -1495,14 +1513,17 @@ func TestComplexForking(t *testing.T) {
...
 	cst2, err := createConsensusSetTester("TestComplexForking - 2")
 	if err != nil {
 		t.Fatal(err)
 	}
+	defer cst2.cs.Close()
 	cst3, err := createConsensusSetTester("TestComplexForking - 3")
 	if err != nil {
 		t.Fatal(err)
 	}
+	defer cst3.cs.Close()
```
Comment (DavidVorick) [inline modules/consensus/accept_test.go]: ditto for cst2 and cst3

File: modules/consensus/consensusset_test.go [hunk 4]
```diff
@@ -87,6 +87,11 @@ func createConsensusSetTester(name string) (*consensusSetTester, error) {
 	return cst, nil
 }
 
+// closeCst safely closes the consensus set tester. 'close' is a builtin
+func (cst *consensusSetTester) closeCst() error {
+	return cst.gateway.Close()
+}
```
Comment (DavidVorick) [inline modules/consensus/consensusset_test.go]: @lukechampine 

Should instead this be `cst.cs.Close()` and then the consensus set calls `gateway.Close()`? Or do you think that modules should not close their dependencies.

I guess modules should not close their dependencies.

File: modules/consensus/accept_test.go (hunk 1 above)
Comment (triazo) [inline modules/consensus/accept_test.go]: It looks like all the places where this function is called have error handling, so I can break out by returning an error.

**Pull request merged**